src/Hooks/useLocalStorage.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react';
+import getToken from '../lib/tokens';
+
+export const useLocalStorage = (key, initialValue) => {
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
+    } catch (error) {
+      return initialValue;
+    }
+  });
+
+  const setValue = (value) => {
+    try {
+      setStoredValue(value);
+      window.localStorage.setItem(key, JSON.stringify(value));
```

Comment:
I like it,  that could you explain me.... is great !!! 🥇 . Congratulations guys.